xarray/coding/variables.py

```diff
@@ -212,11 +212,31 @@ class CFScaleOffsetCoder(VariableCoder):
         decode_values = encoded_values * scale_factor + add_offset
     """
 
+    @staticmethod
+    def _choose_float_dtype(data, has_offset):
```

Comment:
Can you make this helper function use `dtype` as an argument instead of `data`? That would make it clearer that the result dtype is not data dependent.